codegen.js

```diff
@@ -0,0 +1,22 @@
+module.exports = {
+  schema: [
+    {
+      "http://localhost:9090/graphql/query": {
+        headers: {
+          cookie: process.env.REACT_APP_GQL_COOKIE,
```

Comment:
Does graphql-codegen support passing a schema file? Having it hit the introspection endpoint and hacking through auth like this will make it annoying to work with, since you have to have the evergreen server running, and you'll have to update this whenever you do something with your user

Reply:
hitting the introspection endpoint is certainly the way to go for this but i understand what john's saying. we could pass it the path to the local schema file in evergreen. or we could not require authentication for gql endpoint if ENV = dev

Reply:
So the latter idea i gave of disabling auth for dev won't work. what about an auth-less endpoint that only returns the schema?

Reply:
What's the advantage of creating an endpoint to return the file if you have the file already?

Reply:
Also im not sure how much of an issue having the evergreen server running would be chances are if you are regenerating the types you are probably already doing development and need the server running anyways.

Reply:
The schema file is expected to exist as "sdlschema" in the spruce folder for query linting: https://github.com/evergreen-ci/spruce/commit/bc288b6c33ce4f289d5bb1b9468abe3bb2e8cbb1#diff-04c6e90faac2675aa89e2176d2eec7d8R26

Reply:
The schema file and introspection file aren't the same. `graphql-codegen` supports importing a introspection json file. But i cant find a way to import a plain  `.graphql` schema file.

Reply:
i'm on board with what john's saying. you can put the path to the local schema file as the value to the schema key in the config file. https://graphql-code-generator.com/docs/getting-started/schema-field#graphql-file

Reply:
Sounds good to me. Looks like the file has to be named  with a `.graphql` filename for graphql-codegen to pick it up